### ISSUE TITLE
SliverAppBar with `floating: true` does not reappear while scrolling up on desktop

### ISSUE THREAD

**Issue by madhavarshney** [flutter/flutter#60670]
Hello there! It seems that `SliverAppBar` with `floating` set to `true` does not reappear when you scroll back up with your mouse on desktops (tested on web and macOS). It does work, however, if you drag the list (by clicking and dragging your mouse). I'm not sure whether this is the intended behavior or whether this has been reported already as I couldn't find an issue documenting this. Thanks! 

## Steps to Reproduce

1. Go to https://flutter.dev/docs/cookbook/lists/floating-app-bar on your desktop
2. Scroll to the interactive example
3. Scroll down in the list
4. Slowly scroll up (the AppBar does not reappear)
5. Click and drag the list to scroll up (the AppBar does reappear).

**Expected results:** 

The widget SliverAppBar should reappear while scrolling with your mouse.

**Actual results:** 

The widget SliverAppBar does not reappear.

<details>
  <summary>Logs</summary>

```
```

```
```

```
```

</details>

**Comment by madhavarshney** [comment]
Additionally, a similar issue seems to apply to other such behaviors on desktop where dragging to scroll triggers an effect, but using the mouse / trackpad to scroll does not. For example, on macOS in a ListView, when you scroll before the top of the list or after the list, there are no bounce mechanics. On the other hand, when you drag the list to do the same, the list is scrollable even after the end of its items and bounces back. I'm not sure whether this is the intended behavior, but at least on the macOS App Store, the scroll bounce mechanics are applied, so I think they should be implemented in Flutter.

![flutter-scroll-macos-bug](https://user-images.githubusercontent.com/40002855/86213310-5bb99480-bb2e-11ea-8b10-fef227852797.gif)

**Comment by madhavarshney** [comment]
Just did some research on the overscroll behavior on macOS, and it seems like the issues above will be solved by #23604. If that's the case, feel free to close this issue. Thanks!

**Comment by TahaTesser** [comment]
Hi @madhavarshney 
I can reproduce the issue using external mouse and trackpad on `web` and `macos desktop`. You're right when we click on the list and drag, it works on desktop just mobile 


<details> 
<summary>Code Sample</summary> 

``` 
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  MyApp({Key key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final title = 'Floating App Bar';

    return MaterialApp(
      title: title,
      home: Scaffold(
        // No appbar provided to the Scaffold, only a body with a
        // CustomScrollView.
        body: CustomScrollView(
          slivers: <Widget>[
            // Add the app bar to the CustomScrollView.
            SliverAppBar(
              // Provide a standard title.
              title: Text(title),
              // Allows the user to reveal the app bar if they begin scrolling
              // back up the list of items.
              floating: true,
              pinned: true,
              // Display a placeholder widget to visualize the shrinking size.
              flexibleSpace: Placeholder(),
              // Make the initial height of the SliverAppBar larger than normal.
              expandedHeight: 200,
            ),
            // Next, create a SliverList
            SliverList(
              // Use a delegate to build items as they're scrolled on screen.
              delegate: SliverChildBuilderDelegate(
                // The builder function returns a ListTile with a title that
                // displays the index of the current item.
                (context, index) => ListTile(title: Text('Item #$index')),
                // Builds 1000 ListTiles
                childCount: 1000,
              ),
            ),
          ],
        ),
      ),
    );
  }
}

``` 

</details> 

<details> 
<summary>flutter doctor -v</summary> 

``` 
[✓] Flutter (Channel dev, 1.20.0-2.0.pre, on Mac OS X 10.15.5 19F101, locale
    en-GB)
    • Flutter version 1.20.0-2.0.pre at /Users/tahatesser/Code/flutter_dev
    • Framework revision 15a28159bc (8 days ago), 2020-06-23 04:52:58 -0700
    • Engine revision 91a63d6a44
    • Dart version 2.9.0 (build 2.9.0-19.0.dev 7e72c9ae7e)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.0
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.46.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.1

[✓] Connected device (4 available)
    • SM M305F      • 32003c30dc19668f • android-arm64  • Android 10 (API 29)
    • macOS desktop • macos            • darwin-x64     • Mac OS X 10.15.5
      19F101
    • Web Server    • web-server       • web-javascript • Flutter Tools
    • Chrome        • chrome           • web-javascript • Google Chrome
      83.0.4103.116

• No issues found!
``` 

</details>

**Comment by yjbanov** [comment]
This sounds like a general desktop framework issue (i.e. it's not web-specific), so I'll remove the web label. The desktop label sufficiently describes the issue.

**Comment by YeungKC** [comment]
Could not reproduce on the master branch and fixed at #66039, please close the issue.

cc @TahaTesser @yjbanov @gspencergoog
